internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -201,6 +210,72 @@ func (c DockerCmdClient) Push(ctx context.Context, uri string, w io.Writer, tags
 	return parts[1], nil
 }
 
+func (in *RunOptions) generateRunArguments() []string {
+	args := []string{"run"}
+
+	//Add container name option.
+	if in.ContainerName != "" {
+		args = append(args, "--name", in.ContainerName)
+	}
+
+	// Add port mappings.
+	for hostPort, containerPort := range in.ContainerPorts {
+		args = append(args, "-p", fmt.Sprintf("%s:%s", hostPort, containerPort))
+	}
+
+	// Add network option if it's not a "pause" container.
+	if in.ContainerName != "pause" {
+		args = append(args, "--network", "container:pause")
+	}
+
+	//Add secrets as environment variables.
+	for key, value := range in.Secrets {
+		args = append(args, "-e", fmt.Sprintf("%s=%s", key, value))
+	}
+
+	//Add environment variables.
+	for key, value := range in.EnvVars {
+		args = append(args, "-e", fmt.Sprintf("%s=%s", key, value))
+	}
+
+	//Add the image name.
+	if in.ImageURI != "" {
+		args = append(args, in.ImageURI)
+	}
+
+	//Add "sleep infinity" command for the "pause" container
+	if in.ContainerName == "pause" {
+		args = append(args, "sleep", "infinity")
```

Comment:
it feels like we should add a `Command` option to `RunOptions` that let's the caller specify what command to run, instead of choosing it based on the container name